UI/window-basic-main-profiles.cpp

```diff
@@ -985,3 +989,178 @@ void OBSBasic::CheckForSimpleModeX264Fallback()
+		return "NVENC check process output read failure, exit code: " +
+		       std::to_string(exit_code);
+	}
+
+	/* Read devices and attempt to find reason for failure */
+	auto numDevices = config_get_uint(config, "general", "cuda_devices");
+	if (!numDevices)
+		return "No NVIDIA GPUs found";
+
+	/* Check device generation (architecture) */
+	for (uint64_t i = 0; i < numDevices; i++) {
+		std::string section = "device." + std::to_string(i);
+
+		if (!config_has_user_value(config, section.c_str(), "name"))
+			continue;
+
+		const std::string_view name =
+			config_get_string(config, section.c_str(), "name");
+		const std::string_view arch = config_get_string(
+			config, section.c_str(), "architecture_name");
+
+		if (arch == "Kepler") {
+			std::string ret =
+				"NVIDIA Kepler (GTX 600/700 series) GPUs are no longer supported. Your GPU: ";
+			ret += name;
+
+			return ret;
+		}
+	}
+
+	const std::string_view reason =
+		config_get_string(config, "general", "reason");
+
+	/* All generic errors */
+	if (!reason.empty()) {
+		std::string ret = "NVENC check failed with reason: ";
+
+		for (auto &[code, desc] : nvencFailureReasons) {
+			if (reason.find(code) == std::string_view::npos)
+				continue;
+
+			ret += desc;
+			ret += " (code: ";
+			ret += code;
+			ret += ")";
+
+			break;
+		}
+
+		return ret;
+	}
+
+	return std::nullopt;
+}
+
+void OBSBasic::CheckForAdvancedModeEncoders()
+{
+	std::unordered_set<std::string> available_encoders;
+
+	const std::string curStreamEncoder =
+		config_get_string(activeConfiguration, "AdvOut", "Encoder");
+	const std::string curRecEncoder =
+		config_get_string(activeConfiguration, "AdvOut", "RecEncoder");
+
+	size_t idx = 0;
+	const char *id;
+	while (obs_enum_encoder_types(idx++, &id))
+		available_encoders.insert(id);
+
+	// ToDo translate everything here
+
+	auto is_encoder_missing =
+		[&](const std::string &encoder) -> std::optional<std::string> {
+		if (available_encoders.count(encoder))
+			return std::nullopt;
+
+		std::string reason = "Unknown";
+		if (encoder.find("nvenc") != std::string::npos) {
+			if (auto res = CheckNVENCVersion())
+				reason = *res;
+		}
+
+		return reason;
+	};
+
+	auto streamEncError = is_encoder_missing(curStreamEncoder);
+	auto recEncError = is_encoder_missing(curRecEncoder);
+
+	// Show warning message
+	if (streamEncError || recEncError) {
+		QString encoderMissingMessage =
+			"The following encoders are missing:\n\n";
+		const QString msgTemplate = "- %1 encoder (%2): %3\n";
+
+		if (streamEncError) {
+			encoderMissingMessage +=
+				msgTemplate.arg("Streaming")
+					.arg(curStreamEncoder.data())
+					.arg(QString::fromStdString(
+						*streamEncError));
+		}
+
+		if (recEncError) {
+			encoderMissingMessage +=
+				msgTemplate.arg("Recording")
+					.arg(curRecEncoder.c_str())
+					.arg(QString::fromStdString(
+						*recEncError));
+		}
+
+		OBSMessageBox::warning(this, "Encoders missing!",
+				       encoderMissingMessage, false);
```

Comment:
I'd prefer a message box title like "Missing Encoder" (I hate how we currently ignore UX guidelines for these).

Reply:
All the strings here should be considered very non-final since they'll have to go through translation later on.